packages/sonarwhal/src/lib/parsers/babel-config/babel-config.ts

```diff
@@ -0,0 +1,96 @@
+import * as path from 'path';
+import * as ajv from 'ajv';
+
+import { IScanEnd, IFetchEnd, Parser, BabelConfig, IBabelConfigInvalid, IBabelConfigParsed, IBabelConfigInvalidSchema } from '../../types';
+import { Sonarwhal } from '../../sonarwhal';
+import { loadJSONFile } from '../../utils/misc';
+
+
+export default class BabelConfigParser extends Parser {
+    private configFound: boolean = false;
+    private schema: any;
+
+    public constructor(sonarwhal: Sonarwhal) {
+        super(sonarwhal);
+
+
+        this.schema = loadJSONFile(path.join(__dirname, 'schema', 'babelConfigSchema.json'));
+        /**
+         * .babelrc => type: 'unknown' ('file-type' module doesn't support the type of 'json').
+         * babelrc.json => type: 'json' (file type from extention).
+         */
+        sonarwhal.on('fetch::end::json', this.parseBabelConfig.bind(this));
+        sonarwhal.on('fetch::end::unknown', this.parseBabelConfig.bind(this));
+        sonarwhal.on('scan::end', this.parseEnd.bind(this));
+    }
+
+    private async parseEnd(scanEnd: IScanEnd) {
+        if (!this.configFound) {
+            await this.sonarwhal.emitAsync('notfound::babel-config', scanEnd);
+        }
+    }
+
+    private async validateSchema(config: BabelConfig, resource: string) {
+        // ajv is lower case to be able to get the types.
+        const x: ajv.Ajv = new ajv({ // eslint-disable-line new-cap
```

Comment:
Why can't we do `new Ajv` ?

Reply:
if you use Ajv as name for the variable, you can't use `ajv` types.